auth/src/main/java/org/aerogear/mobile/security/SecurityCheckExecutorListener.java

```diff
@@ -0,0 +1,20 @@
+package org.aerogear.mobile.security;
+
+import org.aerogear.mobile.security.SecurityCheckResult;
+
+/**
+ * Listener for events about check execution.
+ */
+public interface SecurityCheckExecutorListener {
+
+    /**
+     * Called after each check is executed
+     * @param result the result of the check
+     */
+    void onExecuted(SecurityCheckResult result);
+
+    /**
+     * Called when all submitted checks has been executed.
+     */
+    void onFinished();
```

Comment:
I'd call it `onComplete` to be compliant with RxJava style programming

Reply:
👍

Reply:
what's happening onError ?

Reply:
Currently checks return just `passed` (true/false) and don't throw any exception.

Reply:
Would it be too much trouble passing all the `results` here?

Reply:
@paolobueno It wouldn't, but I don't like the idea of caching all the results in the executor. Currently the results are very small objects and the number of checks is small, but we can never know what is going to happen in the future.
Better to pass them one by one and let the receiver decide what to do with them